public/app/core/components/colorpicker/withColorPicker.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Drop from 'tether-drop';
+import { SeriesColorPicker } from './SeriesColorPicker';
+
+export interface WithSeriesColorPickerProps {
+  color: string;
+  yaxis?: number;
+  optionalClass?: string;
+  onColorChange: (newColor: string) => void;
+  onToggleAxis?: () => void;
+}
+
+export default function withSeriesColorPicker(WrappedComponent) {
```

Comment:
Why is this class wrapped by a function? Could you not render this.props.children below instead?

Reply:
@alexanderzobnin if you are using HOC here, it's wise to give it a `displayName` for increased readability when debugging. More on it here https://reactjs.org/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging